test_conformance/api/test_queries.cpp

```diff
@@ -244,79 +249,203 @@ int test_get_sampler_info(cl_device_id deviceID, cl_context context, cl_command_
-    test_error( error, "Unable to get command queue reference count" );
-    if( size != sizeof( refCount ) )
-    {
-        log_error( "ERROR: Returned size of command queue reference count does not validate! (expected %d, got %d)\n", (int)sizeof( refCount ), (int)size );
-        return -1;
-    }
+    CL_QUEUE_ON_DEVICE | CL_QUEUE_OUT_OF_ORDER_EXEC_MODE_ENABLE,
 
-    cl_context otherCtx;
-    TEST_COMMAND_QUEUE_PARAM( queue, CL_QUEUE_CONTEXT, otherCtx, context, "context", "%p", cl_context )
+    CL_QUEUE_PROFILING_ENABLE | CL_QUEUE_ON_DEVICE
+        | CL_QUEUE_OUT_OF_ORDER_EXEC_MODE_ENABLE,
 
-    cl_device_id otherDevice;
-    error = clGetCommandQueueInfo( queue, CL_QUEUE_DEVICE, sizeof(otherDevice), &otherDevice, &size);
-    test_error(error, "clGetCommandQueue failed.");
+    CL_QUEUE_ON_DEVICE | CL_QUEUE_ON_DEVICE_DEFAULT
+        | CL_QUEUE_OUT_OF_ORDER_EXEC_MODE_ENABLE,
 
-    if (size != sizeof(cl_device_id)) {
-        log_error( " ERROR: Returned size of command queue CL_QUEUE_DEVICE does not validate! (expected %d, got %d)\n", (int)sizeof( otherDevice ), (int)size );
-        return -1;
-    }
+    CL_QUEUE_PROFILING_ENABLE | CL_QUEUE_ON_DEVICE | CL_QUEUE_ON_DEVICE_DEFAULT
+        | CL_QUEUE_OUT_OF_ORDER_EXEC_MODE_ENABLE
+};
+
+int check_get_command_queue_info_params(cl_device_id deviceID,
+                                        cl_context context,
+                                        bool is_compatibility)
+{
+    int error;
+    size_t size;
+
+    cl_queue_properties host_queue_props, device_queue_props;
+    cl_queue_properties queue_props[] = { CL_QUEUE_PROPERTIES, 0, 0 };
+
+    clGetDeviceInfo(deviceID, CL_DEVICE_QUEUE_ON_HOST_PROPERTIES,
+                    sizeof(host_queue_props), &host_queue_props, NULL);
+    log_info("CL_DEVICE_QUEUE_ON_HOST_PROPERTIES is %d\n",
+             (int)host_queue_props);
+    clGetDeviceInfo(deviceID, CL_DEVICE_QUEUE_ON_DEVICE_PROPERTIES,
+                    sizeof(device_queue_props), &device_queue_props, NULL);
+    log_info("CL_DEVICE_QUEUE_ON_HOST_PROPERTIES is %d\n",
+             (int)device_queue_props);
 
-    /* Since the device IDs are opaque types we check the CL_DEVICE_VENDOR_ID which is unique for identical hardware. */
-    cl_uint otherDevice_vid, deviceID_vid;
-    error = clGetDeviceInfo(otherDevice, CL_DEVICE_VENDOR_ID, sizeof(otherDevice_vid), &otherDevice_vid, NULL );
-    test_error( error, "Unable to get device CL_DEVICE_VENDOR_ID" );
-    error = clGetDeviceInfo(deviceID, CL_DEVICE_VENDOR_ID, sizeof(deviceID_vid), &deviceID_vid, NULL );
-    test_error( error, "Unable to get device CL_DEVICE_VENDOR_ID" );
+    auto version = get_device_cl_version(deviceID);
 
-    if( otherDevice_vid != deviceID_vid )
+    // Are on device queues supported
+    bool on_device_supported =
+        (version >= Version(2, 0)
+         || (version >= Version(3, 0) && sizeof(device_queue_props) != 0));
```

Comment:
Still not quite right - the left-hand side of the `||` will be `true` for any 3.0 device, so the overall expression will always be `true` for 3.0.  Also, the check for device queues on 3.0 shouldn't be testing `sizeof()`,

So I think this needs to be `(version >= Version(2, 0) && version < Version(3, 0)) || (Version >= Version(3, 0) && device_queue_props != 0)`. Not pleasant, but it will be made nicer when Kevin's feature testing framework lands.